### PR TITLE
feat: type - add first case

### PR DESCRIPTION
**Pull request by tony-go**
Hi 👋 there!

The purpose of this PR is to tackle the issue #1 ^^

### CODE REVIEW

**Comment by tony-go** [comment]
I only focus on the first case ATM.

File: src/spyOn.ts [hunk 1]
```diff
@@ -28,12 +28,16 @@ export function spyOn<Obj extends object, Method extends Methods<Obj>>(
   obj: Obj,
   methodName: Method,
   mock?: Obj[Method]
-): Spy<Parameters<Obj[Method]>, ReturnType<Obj[Method]>>
+): Obj[Method] extends (...args: Array<any>) => any
```
Comment (sheremet-va) [inline src/spyOn.ts:31]: I think we can use `infer` here:
```
(...args: infer Args) => infer Returns
```

to pass them to `Spy`

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Comment by sheremet-va** [comment]
Also I think that to make the size even smaller, we can make every generic type as a one letter word:
- Obj -> O, Method -> M
And so on

File: src/spyOn.ts (hunk 1 above)
Comment (tony-go) [reply src/spyOn.ts:31]: Don't really get you on this. Could you elaborate?

Comment (tony-go) [reply src/spyOn.ts:31]: Not really familiar with infer probably

Comment (sheremet-va) [reply src/spyOn.ts:31]:
```
O[M] extends (...args: infer A) => R ? Spy<A, R> : never
```

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: src/spyOn.ts (hunk 1 above)
Comment (tony-go) [reply src/spyOn.ts]: we also had to infer R

**Comment by tony-go** [comment]
the size 🙈

**Pull request merged**